### DISCUSSION TITLE
i18n composition api issue

### DISCUSSION THREAD

**Discussion by BalazsMakrai** [quasarframework/quasar#14743] [General - Components / Directives / etc]
I'm trying to use i18n with composition api. The script contains these codes:
```
import { useI18n } from 'vue-i18n';
setup() {
  const { t } = useI18n();
  let message = t('failed');
}
```

After that I get this error in my console:

> Uncaught (in promise) SyntaxError: Not available in legacy mode (at message-compiler.mjs?f4eb:54:1)
>     at createCompileError (message-compiler.mjs?f4eb:54:1)
>     at createI18nError (vue-i18n.mjs?9ca6:92:1)
>     at useI18n (vue-i18n.mjs?9ca6:2223:1)
>     at setup (MailList.vue?915a:18:1)
>     at callWithErrorHandling (runtime-core.esm-bundler.js?f781:155:1)
>     at setupStatefulComponent (runtime-core.esm-bundler.js?f781:7165:1)
>     at setupComponent (runtime-core.esm-bundler.js?f781:7120:1)
>     at mountComponent (runtime-core.esm-bundler.js?f781:5473:1)
>     at processComponent (runtime-core.esm-bundler.js?f781:5448:1)
>     at patch (runtime-core.esm-bundler.js?f781:5038:1)

What's going on here? How could I make this work?

**Comment by MilosPaunovic** [comment]
Have you followed the official guide for implementation?
https://quasar.dev/options/app-internationalization#introduction

**Reply by BalazsMakrai** [reply]
Yes, but I still getting these errors

**Reply by BalazsMakrai** [reply]
I've added this code to ibbot/i18n.js and solved the problem:  ` allowComposition: true`
